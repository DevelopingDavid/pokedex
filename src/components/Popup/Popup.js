import React, { Component } from 'react';

export default class Popup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      extraDetails: {},
      loading: true
    }
  }

  async componentDidMount() {
    let data = await this.grabDetails(this.props.id);
    this.setState({ extraDetails: data, loading: false });
  }

  grabDetails = async (id) => {
    const response = await fetch(`https://pokeapi.co/api/v2/pokemon-species/${id}`);
    const data = await response.json();
    return data;
  }

  render() {
    const { currentPokemon } = this.props;
    const { extraDetails } = this.state;
    let englishText;
    if(Object.keys(extraDetails).length > 0) {
      englishText = extraDetails.flavor_text_entries.find(lang => {
       return lang.language.name === 'en'
      });
    }

    return (
      <article>
        { !this.state.loading &&
          <div>
            <h1>{currentPokemon.name}</h1>
            <img src={currentPokemon.sprite} />
            <h1>weight: {currentPokemon.weight} hectograms</h1>
            <h1>height: {currentPokemon.height} decimetres</h1>
            <h1>kind: {extraDetails.shape.name}</h1>
            <h1>habitat: {extraDetails.habitat.name}</h1>
            <h1>{englishText.flavor_text}</h1>
          </div>
        }
      </article>
    )
  }
};