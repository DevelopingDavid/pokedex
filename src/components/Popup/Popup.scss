.popup-container {
  display: flex;
  justify-content: center;
  margin-top: 60px;
}

.popup-card {
  font-family: "Flexo-Medium",arial,sans-serif;
  display: flex;
  align-items: center;
  border: 1px solid black;
  border-radius: 12px;
  width: 800px;
  background: #FFFFFF;
}

.sprite-image {
  width: 300px;
  height: auto;
  background: #F2F2F2;
  border-radius: 12px;
  background-position:-1500px -1350px;
  margin: 15px;
}

.popup-pokemon{
  div {
    display: flex;
    h1 {
      margin: 10px 0 0 18px;
    }
  }
}

.popup-info {
  width: 500px;
  h1 {
    margin: 25px 5px;
  }
  a {
    text-decoration: none;
    color: black;
    font-size: 1.4rem;
    &:hover {
      color: #D8000F;
    }
  }
}