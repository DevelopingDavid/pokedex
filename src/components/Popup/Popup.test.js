import React from 'react';
import { shallow } from 'enzyme';
import Popup from './Popup';
let mockpokemon = {
  dexNumber: 1,
  height: 7,
  name: "bulbasaur",
  sprite: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png",
  types: (2)[{}, {}],
  weight: 69
}
describe('Popup', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Popup id={1} currentPokemon={mockpokemon}/>
    )
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call upon grabDetails on component`DidMount', () => {
    let instance = wrapper.instance();
    let spy = jest.spyOn(instance, 'grabDetails');
    instance.componentDidMount();
    expect(spy).toHaveBeenCalled();
  });

  it('should setState on componentDidMount', async () => {
    let instance = wrapper.instance();
    await instance.componentDidMount();
    expect(wrapper.state('loading')).toEqual(false)
  });
});