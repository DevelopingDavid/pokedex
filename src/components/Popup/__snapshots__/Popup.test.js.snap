// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup should match snapshot 1`] = `
<article
  className="popup-container"
/>
`;
